src/utils.js

```diff
@@ -54,13 +54,20 @@ function countryCodeToGoogleHostname(countryCode) {
     return googleDomains[suffix];
 }
 
+// New function which forms a URL from countryCode and query params
+function formUrl(countryCode, query) {
+    const hostname = countryCodeToGoogleHostname(countryCode);
+    const url = `http://www.${hostname}/search?q=${encodeURIComponent(query)}&tbm=shop&tbs=vw:l`;
+    return { url, hostname };
+}
+
 async function makeRequestList(queries, inputUrl, countryCode) {
     const hostname = countryCodeToGoogleHostname(countryCode);
-    let sources;
+    let sources = [];
 
     if (!inputUrl) {
         sources = queries.map((query) => {
-            const url = `http://www.${hostname}/search?q=${encodeURIComponent(query)}&tbm=shop&tbs=vw:l`;
+            const { url } = formUrl(countryCode, query);
```

Comment:
`hostname` is now coming from `formUrl` or it should always be using the topmost hostname?

Reply:
hostname is used within formUrl